### PR TITLE
Add, list, read and delete files.

### PR DESCRIPTION
**Pull request by howbazaar**
Tests involved some http request wrangling, but it's all good now.

### CODE REVIEW

File: interfaces.go [hunk 1]
```diff
@@ -49,6 +49,30 @@ type Controller interface {
...
+	// file without sending the content of the file, we can return a File
+	// instance here too.
+	AddFile(AddFileArgs) error
+}
+
+// File represents a file stored in the MAAS controller.
+type File interface {
+	// Filename is the name of the file. No path, just the filename.
+	Filename() string
+	// AnonymousURL is a URL that can be used to retrieve the conents of the
```
Comment (mjs) [inline interfaces.go:73]: blank line

File: interfaces.go [hunk 2]
```diff
@@ -49,6 +49,30 @@ type Controller interface {
...
+	// file without credentials.
+	AnonymousURL() string
+
+	// Delete removes the file from the MAAS controller.
+	Delete() error
+	// ReadAll returns the content of the file.
```
Comment (mjs) [inline interfaces.go:80]: blank line

**Comment by mjs** [comment]
LGTM

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**